### PR TITLE
Adds PREPARE and DEALLOCATE

### LINKED ISSUE #101: Implement PREPARE and DEALLOCATE statement
The following specs exists which bypasses the arel visistor

```ruby
  visit 'sql',
        'PREPARE some_plan (integer) AS (SELECT $1)',
        pg_node: 'PgQuery::PREPARE_STMT',
        sql_to_arel: false
```

```ruby
  visit 'sql', 'DEALLOCATE some_prepared_statement',
        pg_node: 'PgQuery::DEALLOCATE_STMT',
        sql_to_arel: false

```

These statements need implementing because these are executed a lot by ActiveRecord.

### PR DESCRIPTION
**Pull request by Willianvdv**
Fixes #101

### CODE REVIEW

File: lib/arel/extensions/prepare_statement.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# rubocop:disable Naming/MethodName
+# rubocop:disable Naming/UncommunicativeMethodParamName
+
+module Arel
+  module Visitors
+    class Dot
+      alias visit_Arel_Nodes_Dealocate terminal
+      alias visit_Arel_Nodes_Prepare terminal
```
Comment (Willianvdv) [inline lib/arel/extensions/prepare_statement.rb]: These visitors are something I guessed and am not exactly what these do yet

File: spec/arel/sql_to_arel_spec.rb [hunk 2]
```diff
@@ -102,6 +98,13 @@
         sql_to_arel: false
   visit 'sql', 'EXPLAIN SELECT 1', pg_node: 'PgQuery::EXPLAIN_STMT', sql_to_arel: false
   visit 'sql', 'FETCH some_cursor', pg_node: 'PgQuery::FETCH_STMT', sql_to_arel: false
+  visit 'sql',
+        'PREPARE some_plan (integer) AS (SELECT $1)',
+        pg_node: 'PgQuery::PREPARE_STMT',
+        sql_to_arel: true
```
Comment (mvgijssel) [inline spec/arel/sql_to_arel_spec.rb]:
```suggestion
```

File: spec/arel/sql_to_arel_spec.rb [hunk 3]
```diff
@@ -102,6 +98,13 @@
...
+  visit 'sql', 'DEALLOCATE some_prepared_statement',
+        pg_node: 'PgQuery::DEALLOCATE_STMT',
+        sql_to_arel: true
```
Comment (mvgijssel) [inline spec/arel/sql_to_arel_spec.rb]:
```suggestion
```

File: spec/arel/sql_to_arel_spec.rb [hunk 4]
```diff
@@ -102,6 +98,13 @@
         sql_to_arel: false
   visit 'sql', 'EXPLAIN SELECT 1', pg_node: 'PgQuery::EXPLAIN_STMT', sql_to_arel: false
   visit 'sql', 'FETCH some_cursor', pg_node: 'PgQuery::FETCH_STMT', sql_to_arel: false
+  visit 'sql',
+        'PREPARE some_plan (integer) AS (SELECT $1)',
```
Comment (mvgijssel) [inline spec/arel/sql_to_arel_spec.rb:102]: Could you add more specs, which also test:
- no arguments for the PREPARE
- DEALLOCATE ALL (https://www.postgresql.org/docs/10/sql-deallocate.html)

File: lib/arel/extensions/prepare_statement.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
+# rubocop:disable Naming/MethodName
```
Comment (mvgijssel) [inline lib/arel/extensions/prepare_statement.rb]: Could you make two separate files `lib/arel/extensions/dealocate.rb` and `lib/arel/extensions/prepare.rb`?

**Review by mvgijssel — APPROVED** [review]

File: lib/arel/extensions.rb [hunk 6]
```diff
@@ -107,6 +107,8 @@
 require 'arel/extensions/false'
 require 'arel/extensions/true'
 require 'arel/extensions/to_sql'
+require 'arel/extensions/prepare'
```
Comment (Willianvdv) [inline lib/arel/extensions.rb:110]: @mvgijssel u happy now?

Comment (mvgijssel) [reply lib/arel/extensions.rb:110]: Yes

**Review by mvgijssel — APPROVED** [review]

**Pull request merged**